e2e/test_playwright.py

```diff
@@ -0,0 +1,337 @@
+import pytest
+from piccolo_admin.example import Director
+from piccolo_admin.example_data import DIRECTORS
+from playwright.sync_api import Playwright, sync_playwright
+
+
+@pytest.fixture
+def reset_database() -> None:
+    Director.delete(force=True).run_sync()
+    Director.insert(*[Director(**d) for d in DIRECTORS]).run_sync()
+
+
+def test_login_logout(playwright: Playwright, reset_database) -> None:
+    browser = playwright.chromium.launch()
+    context = browser.new_context()
+    # Open new page
+    page = context.new_page()
+    # Go to http://localhost:3000/#/login?nextURL=%2F
+    page.goto("http://localhost:3000/#/login?nextURL=%2F")
```

Comment:
Out of interest, did you generate these using Playwright Codegen?

If so, how did you find it?

Reply:
Sorry for the late reply. Yes, tests are generated from `playwright codegen localhost:3000`. After that, two windows were opened. One where you can interact with the web page and one where the code is generated. I had to change each test a bit later, but the tests are 80% generated with Playwright and I think it's a great feature that is very useful.